src/js/api.ts

```diff
@@ -40,7 +40,7 @@ API.setPyProxyToStringMethod = function (useRepr: boolean): void {
   Module.HEAP8[Module._compat_to_string_repr] = +useRepr;
 };
 
-/** @private */
+/** @hidden */
```

Comment:
Are we replacing `@private` decorator to `@hidden` at all? Or are there some difference between the two?

Reply:
`@hidden` makes typedoc drop the object from the ir:
https://github.com/TypeStrong/typedoc/blob/master/src/lib/converter/plugins/CommentPlugin.ts#L568

`@private` gets stored into `relection.flags.isPrivate` and gets excluded if the `--excludePrivate` argument is passed to typedoc:
https://github.com/TypeStrong/typedoc/blob/master/src/lib/converter/plugins/CommentPlugin.ts#L500-L505

We don't pass `--excludePrivate`. So instead we decide whether it should be private here by looking at the Python `is_private` field on the TopLevel:
https://github.com/pyodide/pyodide/blob/main/docs/sphinx_pyodide/sphinx_pyodide/jsdoc.py#L87-L90

On TopLevels that can also be "class members" we copy the `isPrivate` flag into the `is_private` ir field. We don't distinguish between a TopLevel function and a class Method, nor a TopLevel variable export and a class attribute. So variables, attributes, functions, and methods are all hidden by `@private` but `TypeAlias`, `Interface`, and `Class` are not.
https://github.com/pyodide/sphinx-js-fork/blob/master/sphinx_js/ir.py?plain=1#L143

Needless to say this requires some cleanup.

Reply:
I'll try to fix this in a followup.